src/rngs/xoshiro128plusplus.rs

```diff
@@ -33,29 +33,36 @@ impl SeedableRng for Xoshiro128PlusPlus {
     /// mapped to a different seed.
     #[inline]
     fn from_seed(seed: [u8; 16]) -> Xoshiro128PlusPlus {
-        if seed.iter().all(|&x| x == 0) {
-            return Self::seed_from_u64(0);
-        }
         let mut state = [0; 4];
         read_u32_into(&seed, &mut state);
+        // Check for zero on aligned integers for better code generation.
+        // Furtermore, seed_from_u64(0) will expand to a constant when optimized.
+        if state.iter().all(|&x| x == 0) {
+            return Self::seed_from_u64(0);
+        }
         Xoshiro128PlusPlus { s: state }
     }
 
     /// Create a new `Xoshiro128PlusPlus` from a `u64` seed.
     ///
     /// This uses the SplitMix64 generator internally.
+    #[inline]
     fn seed_from_u64(mut state: u64) -> Self {
         const PHI: u64 = 0x9e3779b97f4a7c15;
-        let mut seed = Self::Seed::default();
-        for chunk in seed.as_mut().chunks_mut(8) {
+        let mut s = [0; 4];
+        for i in s.chunks_exact_mut(2) {
             state = state.wrapping_add(PHI);
             let mut z = state;
             z = (z ^ (z >> 30)).wrapping_mul(0xbf58476d1ce4e5b9);
             z = (z ^ (z >> 27)).wrapping_mul(0x94d049bb133111eb);
             z = z ^ (z >> 31);
-            chunk.copy_from_slice(&z.to_le_bytes());
+            i[0] = z.to_le() as u32;
+            i[1] = (z.to_le() >> 32) as u32;
```

Comment:
If I remove it, won't the result be different than before on BE? I was trying to preserve that here.

Reply:
There's a `reference` test in the same file, but in this case testing a single sample should be enough. And yes, results should match from prior to your PR.

It doesn't really matter since we explicitly do not promise value stability of these RNGs, but feel like we should anyway.

Reply:
Done in [bee90d1](https://github.com/rust-random/rand/pull/1482/commits/bee90d16dcaebd52094d0bdb8bedf55feef68452) and [4d6b651](https://github.com/rust-random/rand/pull/1482/commits/4d6b65133d4525f5d4f31b4ca10ccda707eb41c9)